File: AlgoPersonalStudy/src/backjoon/silver/BJ_11399.java
package backjoon.silver;

import java.io.*;
import java.util.*;

public class BJ_11399 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] arr = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int n=0;n<N;n++){
            arr[n] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);
        int answer = 0;

        for(int n=0;n<N;n++){
            answer += arr[N-n-1]*(n+1);
        }
        System.out.println(answer);
    }
}

File: AlgoPersonalStudy/src/_kotlin/backjoon/silver/BJ_11399.kt
package _kotlin.backjoon.silver

import java.io.*
import java.util.*

class BJ_11399 {
    fun solution(){
        val br = BufferedReader(InputStreamReader(System.`in`))
        val n = Integer.parseInt(br.readLine())

        val arr = IntArray(n)
        val st = StringTokenizer(br.readLine())
        for (i in 0 until n) {
            arr[i] = st.nextToken().toInt()
        }
        Arrays.sort(arr)
        var answer = 0
        for(i in 0 until n){
            answer += arr[n-i-1] * (i+1)
        }
        println(answer)
    }
}
fun main(){
    val solutionClass = BJ_11399()
    solutionClass.solution()
}